### PR TITLE
Fix JMH dependencies

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
Fix JMH dependencies
==COMMIT_MSG==

### CODE REVIEW

File: crypto-core/build.gradle [hunk 1]
```diff
@@ -11,8 +11,8 @@ dependencies {
     implementation 'com.palantir.safe-logging:preconditions'
     implementation 'com.palantir.safe-logging:safe-logging'
 
-    annotationProcessor 'org.openjdk.jmh:jmh-generator-annprocess'
-    compileOnly 'org.openjdk.jmh:jmh-generator-annprocess'
+    testAnnotationProcessor 'org.openjdk.jmh:jmh-generator-annprocess'
+    testCompileOnly 'org.openjdk.jmh:jmh-generator-annprocess'
```
Comment (carterkozak) [inline crypto-core/build.gradle:15]: The benchmark lives in `src/test/java`

Comment (schlosna) [reply crypto-core/build.gradle:15]: maybe separate, but should we move them to `src/jmh/java` use use [jmh-gradle-plugin](https://github.com/melix/jmh-gradle-plugin)?

**Review by schlosna — COMMENTED** [review]
Approving as this is strictly an improvement, though wondering if we should move the benchmark

**Review by schlosna — APPROVED** [review]

File: crypto-core/build.gradle (hunk 1 above)
Comment (carterkozak) [reply crypto-core/build.gradle:15]: Agree that would be cleaner, I've had some dependency issues with the gradle plugin that are probably solved or irrelevant at this point -- I usually create a separate module instead, but either way would be an improvement over reusing test sources.

**Pull request merged**